### PR TITLE
[TT-8515] Add documentation for GoPlugin ctx.GetOASDefinition

### PR DESCRIPTION
**Pull request by furkansenharputlu**
Just copied and modified the doc for `ctx.GetDefinition` which is one heading above.

### CODE REVIEW

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 1]
```diff
@@ -711,6 +711,33 @@ func MyPluginFunction(w http.ResponseWriter, r *http.Request) {
...
+  "net/http"
+  "github.com/TykTechnologies/tyk/ctx"
+)
+func main() {}
+func MyPluginFunction(w http.ResponseWriter, r *http.Request) {
+  oas := ctx.GetOASDefinition(r)
+  fmt.Println("OAS doc title is", oas.Info.Title)
+}
+```
+`ctx.GetOASDefinition` returns an OAS object, the Go data structure can be found [here](https://github.com/TykTechnologies/tyk/blob/master/apidef/oas/oas.go#L25)
```
Comment (andyo-tyk) [inline tyk-docs/content/plugins/supported-languages/golang.md]:
```suggestion
`ctx.GetOASDefinition` returns an `OAS` object containing the Tyk OAS API definition. The Go data structure can be found [here](https://github.com/TykTechnologies/tyk/blob/master/apidef/oas/oas.go#L25)
```

File: tyk-docs/content/plugins/supported-languages/golang.md [hunk 2]
```diff
@@ -711,6 +711,33 @@ func MyPluginFunction(w http.ResponseWriter, r *http.Request) {
...
+
+```
+{{< warning success >}}
+**Warning**
+
+`ctx.GetDefinition` returns `nil` for Tyk OAS APIs and `ctx.GetOASDefinition` returns `nil` for Tyk Classic APIs.
```
Comment (andyo-tyk) [inline tyk-docs/content/plugins/supported-languages/golang.md]:
```suggestion
`ctx.GetDefinition` returns `nil` if called from a Tyk OAS API and `ctx.GetOASDefinition` returns `nil` if called from a Tyk Classic API.
```

**Review by andyo-tyk — CHANGES_REQUESTED** [review]
Minor changes to improve clarity

**Review by andyo-tyk — APPROVED** [review]
Great, thanks :)

**Review by lghiur — APPROVED** [review]

**Review by dcs3spp — APPROVED** [review]
lgtm

**Pull request merged**